specs/lib/combinePrimitivesSpec.js

```diff
@@ -1,99 +1,84 @@
-        oneAttribute.meshes.meshTest.primitives[0].attributes = {
-            "POSITION": 'accessor_58'
-        };
-        combinePrimitives(oneAttribute);
-        expect(oneAttribute.meshes.meshTest.primitives.length).toEqual(2);
-
-        var differentAttributes = clone(doubleBoxError);
-        differentAttributes.meshes.meshTest.primitives[0].attributes = {
-            "NORMAL": 'accessor_60',
-            "POS": 'accessor_58',
-            "TEXCOORD_0": 'accessor_62'
-        };
-        combinePrimitives(differentAttributes);
-        expect(differentAttributes.meshes.meshTest.primitives.length).toEqual(2);
+        readGltf(doubleBoxToCombinePath, function(gltf){
+            var oneAttribute = gltf;
+
+            oneAttribute.meshes.meshTest.primitives[0].attributes = {
+                "POSITION": 'accessor_58'
+            };
+            combinePrimitives(oneAttribute);
+            expect(oneAttribute.meshes.meshTest.primitives.length).toEqual(2);
+        });
 
+        readGltf(doubleBoxToCombinePath, function(gltf){
+            var differentAttributes = gltf;
+            differentAttributes.meshes.meshTest.primitives[0].attributes = {
+                "NORMAL": 'accessor_60',
+                "POS": 'accessor_58',
+                "TEXCOORD_0": 'accessor_62'
+            };
+            combinePrimitives(differentAttributes);
+            expect(differentAttributes.meshes.meshTest.primitives.length).toEqual(2);
+        });
     });
 
-    it('combines two primitives', function() {
-        combinePrimitives(doubleBoxToCombine);
-        writeGltf(doubleBoxToCombine, './doubleBoxToCombineOutput.gltf', false, true, function() {
+    it('combines two primitives', function(){
+        readGltf(doubleBoxToCombinePath, function(gltf){
+            var doubleBoxToCombine = gltf;
+
+            combinePrimitives(doubleBoxToCombine);
             expect(doubleBoxToCombine.meshes.meshTest.primitives.length).toEqual(1);
             expect(doubleBoxToCombine.meshes.meshTest.primitives[0].material).toEqual('Effect_outer');
             expect(doubleBoxToCombine.meshes.meshTest.primitives[0].mode).toEqual(4);
             expect(doubleBoxToCombine.meshes.meshTest.primitives[0].indices).toEqual('meshTest_INDEX_accessor_0');
+
             expect(doubleBoxToCombine.meshes.meshTest.primitives[0].attributes).toEqual({
                 "NORMAL": 'meshTest_NORMAL_accessor_0',
                 "POSITION": 'meshTest_POSITION_accessor_0',
                 "TEXCOORD_0": 'meshTest_TEXCOORD_0_accessor_0'
             });
+
             expect(doubleBoxToCombine.accessors['meshTest_INDEX_accessor_0']).toEqual({
                 "bufferView": "meshTest_INDEX_bufferView_0",
                 "byteOffset": 0,
                 "byteStride": 0,
                 "componentType": 5123,
                 "type": "SCALAR",
-                "count": 516
+                "count": 516,
+                "extras": { "_pipeline": { "deleteExtras": true } }

```

Comment:
Is there a way to get around checking this, it just seems out of place. Maybe a call to `removePipelineExtras` beforehand?